lib/editor-linter.coffee

```diff
@@ -1,102 +1,41 @@
-{CompositeDisposable, Emitter, Range} = require 'atom'
-Helpers = require './helpers'
+{TextEditor, Emitter, CompositeDisposable} = require('atom')
 
 class EditorLinter
-  constructor: (@linter, @editor) ->
-    @status = true
-    @messages = new Map
-    @inProgress = false
-    @inProgressFly = false
-
-    if @editor is atom.workspace.getActiveTextEditor()
-      @linter.views.updateLineMessages(true)
-
+  constructor: (@editor) ->
+    throw new Error("Given editor isn't really an editor") unless @editor instanceof TextEditor
     @emitter = new Emitter
     @subscriptions = new CompositeDisposable
-
-    @subscriptions.add(
-      @editor.onDidSave => @lint(false)
-    )
-    @subscriptions.add(
-      @editor.onDidChangeCursorPosition ({oldBufferPosition, newBufferPosition}) =>
+    @subscriptions.add @editor.onDidDestroy =>
+      @emitter.emit 'did-destroy'
+
+    @subscriptions.add @editor.onDidSave => @emitter.emit('should-lint', false)
+    # The onDidStopChanging callbacks are invoked immediately, and we want to avoid it
+    setImmediate =>
```

Comment:
The one inside it is to trigger the `should-lint` asynchronously, so the other handlers on that event don't get blocked because of it.